libmysqlconnectorcpp: 8.0.21 -> 8.0.22

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/libmysqlconnectorcpp/versions.


meta.description for libmysqlconnectorcpp is: "C++ library for connecting to mysql servers"

meta.homepage for libmysqlconnectorcpp is: "https://dev.mysql.com/downloads/connector/cpp/"

meta.changelog for libmysqlconnectorcpp is: ""


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 8.0.22 with grep in /nix/store/x0r90bngr6kgsl82h0bfnq52qmblpviv-libmysqlconnectorcpp-8.0.22
- directory tree listing: https://gist.github.com/89baadf77527cda7626168f9b7f2fb94

- du listing: https://gist.github.com/3bbf2d4d2e8f71eaca73842cdbb18e0a

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
7 total rebuild path(s)

2 package rebuild(s)

2 x86_64-linux rebuild(s)
2 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
2 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
libmysqlconnectorcpp
mysql-workbench
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/x0r90bngr6kgsl82h0bfnq52qmblpviv-libmysqlconnectorcpp-8.0.22 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A libmysqlconnectorcpp https://github.com/r-ryantm/nixpkgs/archive/7166c80718c3bc4e640d100691f574eed2285022.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/x0r90bngr6kgsl82h0bfnq52qmblpviv-libmysqlconnectorcpp-8.0.22
ls -la /nix/store/x0r90bngr6kgsl82h0bfnq52qmblpviv-libmysqlconnectorcpp-8.0.22/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libmysqlconnectorcpp</li>
    <li>mysqlWorkbench (mysql-workbench)</li>
  </ul>
</details>

---

###### Maintainer pings

jtojnar reviewed (approved):
Changes look okay: https://dev.mysql.com/doc/relnotes/connector-cpp/en/news-8-0-22.html

Binary distribution now ships plug-ins and and users will have to set path to them as if they want to use LDAP since it cannot really be installed to this package (where it will likely look for plug-ins).

jtojnar committed:
libmysqlconnectorcpp: 8.0.21 -> 8.0.23

- https://dev.mysql.com/doc/relnotes/connector-cpp/en/news-8-0-23.html
- https://dev.mysql.com/doc/relnotes/connector-cpp/en/news-8-0-22.html

Binary distribution now ships plug-ins and and users will have to set path to them as if they want to use LDAP since it cannot really be installed to this package (where it will likely look for plug-ins).

jtojnar commented:
@GrahamcOfBorg build mysql-workbench
